src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -2045,6 +2067,23 @@ GenerateTaskMoveDependencyList(PlacementUpdateEvent *move, int64 colocationId,
 		}
 	}
 
+	*nDepends = hash_get_num_entries(dependsList);
+	if (*nDepends == 0)
+	{
```

Comment:
This is confusing to me - `nDepends` equal to zero meant, no dependent shard tasks?